### ISSUE TITLE
Latex compilation breaks

### ISSUE THREAD

**Issue by iarayad** [executablebooks/mystmd#807]
## Description

I am using a myst template created by me to compile a pdf that includes latex and Python code.

Whenever I compile the document in versions v1.1.29 or newer, I get the error `TypeError: Cannot read properties of undefined (reading 'map')` . In older versions, the pdf compiles just fine.

Unfortunately, I have not been able to produce a minimal example. There is a `.md` file that I include which breaks things down, but I don't know what part exactly. For example,  if I comment out the whole file at once, compilation works, but if I comment it out in "parts", then compilation may still break.

 The file that makes compilation break contains math cells using `:::{math}`, ` ```{math}`, `:::{admonition}`, ` ```{embed}`,  and a markdown list. The files that do not break compilation do not use  `:::{math}`, ` ```{math}`, or  `:::{admonition}`, so the error may be related to these.

Here is an extended version of the log in case it helps:

```
...
📖 Built algorithms.ipynb in 285 ms.
📖 Built benchmark.md in 279 ms.
📖 Built conclusion.md in 278 ms.
📖 Built finding_effective_model.ipynb in 279 ms.
📖 Built implementation.md in 272 ms.
📖 Built scipost-template/python_style.tex in 259 ms.
📖 Built scipost-template/README.md in 258 ms.
📖 Built scipost-template/template.tex in 257 ms.
TypeError: Cannot read properties of undefined (reading 'map')
```

It seems to be an uncaught Javascript error.

## Proposed solution

I don't know.

## Additional notes

The code used to reproduce the error can be found [here](https://gitlab.kwant-project.org/qt/pymablock/-/tree/paper/paper?ref_type=heads) by running `myst build paper.md --pdf`.
The merge request jobs can be found [here](https://gitlab.kwant-project.org/qt/pymablock/-/merge_requests/54).
I am sorry that I cannot provide a shorter failing example.

**Comment by akhmerov** [comment]
For completeness, here's a reproduction script (non-minimal), and assumes that myst 1.1.34 is installed:
```bash
git clone ssh://git@gitlab.kwant-project.org:443/qt/pymablock.git --branch paper
cd pymablock/paper
myst build paper.md
```
**Expected**: `export/paper.pdf` is created
**What actually happens**: no output, uncaught JS error in terminal.

Extra observations:
- Removing inclusion of `algorithms.md` from `paper.md` fixes the build
- Copying `algorithms.md` into `paper.md`, and removing jupytext frontmatter fixes the build.

**Comment by rowanc1** [comment]
Thank you for the report! Have been able to reproduce the bug locally, but haven't figured it out quite yet. @fwkoch can you take a look at this?!

**Comment by fwkoch** [comment]
This was a problem with how `myst` was parsing references with implicit labels, e.g. `` {eq}`unitarity` `` - I fixed it up here: https://github.com/executablebooks/mystmd/pull/808

I'll plan to publish a new release today with this fix.

**Comment by fwkoch** [comment]
Ok - this is released with v1.1.36 - hopefully that works well for you.

Let us know if you still run into problems or have any other questions with your MyST template!
